jedi/evaluate/instance.py

```diff
@@ -24,6 +24,7 @@ def __init__(self, evaluator, parent_context, class_context, var_args):
         super(AbstractInstanceContext, self).__init__(evaluator, parent_context)
         # Generated instances are classes that are just generated by self
         # (No var_args) used.
+        self.tree_node = class_context.tree_node
```

Comment:
Hmm, thanks for the test and fix. However I'm not sure I like this here. If anything I would have probably moved that assignment to `jedi.evaluate.instance.TreeInstance`. Would that make sense for you?